src/preview.cpp

```diff
@@ -506,8 +506,8 @@ namespace {
                 return false;
             image->readMetadata();
 
-            width_ = image->pixelWidth();
-            height_ = image->pixelHeight();
+            width_ = static_cast<uint32_t>(image->pixelWidth());
```

Comment:
Hum ... I am a bit skeptical about this change. I noticed that in your second commit (76d4748d97d3a6cfff113e12298fb20d70336502) you changed pixelWidth and pixelHeight from `int` to `int64_t`, not only in the function members but also in the class members `Image::pixelWidth_` and `Image::pixelHeight_`. 

What's the rationale for addressing these changes? 

In your first commit (256365830a5ce7a917d48acca8ba0b58569f4510) I noticed that after making the change you could get rid of some `static_cast`s. It is always good to get rid of them, their presence signal that something is not modeled correctly.

However in the last commits I noticed that you had to add few `static_cast`s here and there. This make me a bit nervous ... unless you have a good reason for it.

Reply:
You're right, I've been lazy here: swapping `int` with `int64_t` in the header was the easiest way to switch it to a platform-independent type. But `uint32_t` is probably a more appropriate choice for `pixelWidth` and `pixelHeight`.

I'll convert this PR back to draft and work on it a bit more.

Reply:
I added `Metadatum::toUint32()` so that the code doesn't need to do any static casts. The implementation of `Metadatum::toUint32()` currently just does a `static_cast` but we could easily add overflow checking in the future if we want to.